cmd/kubeadm/app/util/cgroupdriver.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"strings"
+
+	utilsexec "k8s.io/utils/exec"
+)
+
+func validateCgroupDriver(driver string) error {
+	if driver != "cgroupfs" && driver != "systemd" {
+		return fmt.Errorf("unknown cgroup driver %q", driver)
+	}
+	return nil
+}
+
+func callDockerInfo() (string, error) {
+	execer := utilsexec.New()
+	out, err := execer.Command("docker", "info").Output()
+	if err != nil {
+		return "", fmt.Errorf("cannot execute 'docker info': %v", err)
+	}
+	return string(out), nil
+}
+
+func getCgroupDriverFromDockerInfo(info string) (string, error) {
+	lineSeparator := ": "
+	prefix := "Cgroup Driver"
+	for _, line := range strings.Split(info, "\n") {
+		if !strings.Contains(line, prefix+lineSeparator) {
+			continue
+		}
+		lineSplit := strings.Split(line, lineSeparator)
+		driver := lineSplit[1]
```

Comment:
Check first that there are two items so we don't panic...

Reply:
so i had this check but removed it.
the first line in the loop handles that
```
if !strings.Contains(line, prefix+lineSeparator) {
```
^ because the `lineSeparator` is always present and the split would always result len() > 1.

i can still add the check?

Reply:
okay, add that as a comment and you're good then :)